Engine/Module/ModuleScene.cpp

```diff
@@ -24,9 +24,20 @@ bool ModuleScene::CleanUp()
 GameObject* ModuleScene::CreateGameObject()
 {
 	std::string created_game_object_name = hierarchy.GetNextGameObjectName();
-	GameObject *created_game_object = root->CreateChild(created_game_object_name);
-	
-	return created_game_object;
+	std::unique_ptr<GameObject> created_game_object = std::make_unique<GameObject>(created_game_object_name);
+	created_game_object->SetParent(root);
+
+	GameObject * created_game_object_ptr = created_game_object.get();
+	game_objects_ownership.emplace_back(std::move(created_game_object));
+	return created_game_object_ptr;
```

Comment:
Could be these something like these?
``` c++
created_game_object->SetParent(root);
game_objects_ownership.emplace_back(std::move(created_game_object));

return created_game_object.get();
```
This way all the operations of the shared pointer are next to each other in the code and also we are saving a copy of `created_game_object_ptr`.

Reply:
you can't do that, because once you move created_game_object inside the vector created_game_object pointer would be nullptr

Reply:
Is this due to `created_game_object` being a `std::unique_ptr`?

Reply:
yep

Reply:
Cool!